extra_data/tests/test_streamer.py

```diff
@@ -6,140 +6,26 @@
-                    # unpacked as list by default.
-                    assert list(v1) == list(v2)
-                else:
-                    assert v1 == v2
-            except AssertionError:
-                raise ValueError('diff: {}{}'.format(path, key), v1, v2)
-
-
-@pytest.fixture(scope="session")
-def server_1():
-    server = ZMQStreamer(4444, maxlen=10, protocol_version='1.0')
-    yield server
-
-
-@pytest.fixture(scope="session")
-def server_2_2():
-    server = ZMQStreamer(5555, maxlen=10, protocol_version='2.2')
-    yield server
-
-
-@pytest.fixture(scope="session")
-def client():
-    client = Client('tcp://localhost:5555')
-    yield client
-
-
-class DummyFrame:
-    """Client._deserialize() now expects the message in ZMQ Frame objects.
-
-    TODO: avoid using a private method from extra_data for tests.
-    """
-
-    def __init__(self, data):
-        self.bytes = data
-        self.buffer = data
-
-
-def test_serialize_1(server_1, client):
-    msg = server_1._serialize(DATA)
-
-    assert isinstance(msg, list)
-    assert len(msg) == 1
-    assert msg[-1] == msgpack.dumps(DATA, use_bin_type=True, default=numpack.encode)
-
-    msg_framed = [DummyFrame(b) for b in msg]
-    data, meta = client._deserialize(msg_framed)
-    compare_nested_dict(data, DATA)
-
-
-def test_serialize_2_2(server_2_2, client):
-    msg = server_2_2._serialize(DATA)
-    assert isinstance(msg, list)
-    assert len(msg) == 6
-
-    m0 = msgpack.loads(msg[0], raw=False)
-    assert m0['source'] == 'XMPL/DET/MOD0'
-    assert m0['content'] == 'msgpack'
-    m2 = msgpack.loads(msg[2], raw=False)
-    assert m2['source'] == 'XMPL/DET/MOD0'
-    assert m2['path'] == 'image.data'
-    assert m2['content'] == 'array'
-
-    m2 = msgpack.loads(msg[4], raw=False)
-    print(m2)
-    assert m2['source'] == 'source1'
-    assert m2['content'] == 'msgpack'
-
-    msg_framed = [DummyFrame(b) for b in msg]
-    data, meta = client._deserialize(msg_framed)
-    compare_nested_dict(data, DATA)
-
-    assert meta['source1']['timestamp.tid'] == 9876543210
-
-
-def test_fill_queue(server_2_2):
-    for i in range(10):
-        server_2_2.feed({str(i): {str(i): i}})
-
-    assert server_2_2._buffer.full()
-    with pytest.raises(Full):
-        server_2_2._buffer.put_nowait({'too much': {'prop': 0}})
 
-    for i in range(10):
-        assert server_2_2._buffer.get()[1] == msgpack.dumps({str(i): i})
+@pytest.fixture(scope='function')
+def server():
+    with ZMQStreamer(2222, maxlen=10) as s:
+        yield s
 
 
-def test_req_rep(server_2_2, client):
-    server_2_2.start()
+def test_req_rep(server):
+    client = Client(server.endpoint)
+    data = {'a': {'b': 1}}
 
-    for i in range(3):
-        server_2_2.feed(DATA)
+    for _ in range(3):
+        server.feed(data)
 
-    for i in range(3):
-        data, metadata = client.next()
-        compare_nested_dict(data, DATA)
+    for _ in range(3):
+        d, metadata = client.next()
+        assert d == data
```

Comment:
I think all the functionality this is testing is now in `karabo_bridge`, right?

Reply:
yes, that's right. There's only a slight difference in input parameters (port number vs interface string). Do you want me to remove the test?